rs/nervous_system/common/validation/src/lib.rs

```diff
@@ -0,0 +1,68 @@
+/// Verifies that the url is within the allowed length, and begins with
+/// `http://` or `https://`. In addition, it will return an error in case of a
+/// possibly "dangerous" condition, such as the url containing a username or
+/// password, or having a port, or not having a domain name.
+pub fn validate_proposal_url(
```

Comment:
I wonder if SNS governance has a similar function that could now be deduplicated?

Reply:
Already uses a shared one - this is just moving the shared one to a smaller crate to avoid needing quite so many dependencies.